toolchain/lower/function_context.h

```diff
@@ -18,10 +19,32 @@ namespace Carbon::Lower {
 // `llvm::Function` definition.
 class FunctionContext {
  public:
-  explicit FunctionContext(FileContext& file_context, llvm::Function* function,
-                           SemIR::SpecificId specific_id,
-                           llvm::DISubprogram* di_subprogram,
-                           llvm::raw_ostream* vlog_stream);
+  explicit FunctionContext(
+      FileContext& file_context, llvm::Function* function,
+      SemIR::SpecificId specific_id, llvm::DISubprogram* di_subprogram,
+      llvm::raw_ostream* vlog_stream,
+      FileContext::SpecificFunctionFingerprint* specific_function_state);
```

Comment:
```suggestion
      SemIR::SpecificId specific_id,
      FileContext::SpecificFunctionFingerprint* specific_function_state,
      llvm::DISubprogram* di_subprogram,
      llvm::raw_ostream* vlog_stream);
```

Sorry, late parameter order nitpick: maybe pass specific parameters adjacent?

Also, can you please add a comment about nullness semantics of specific_function_state?